File: ThingTest/src/ThingTest.java
import java.util.*;

public class ThingTest {
    public static void main(String[] args) {

	ArrayList<Thing> things = new ArrayList<Thing>();
        ArrayList<String> LabelPile;

	String[] MenuOptions = { "Add a [Thing] to your pile.",
	                         "View your pile.",
                                 "Discard a [Thing].", 
	                         "Exit" };

        NumMenu MainMenu = new NumMenu(MenuOptions);

        System.out.print(TuiTeX.box("Things"));

        while(true){			
	    switch(MainMenu.getSelection(true)){

                case 1:
    	            things.add(new Thing());
                    break;		    		

	        case 2:
                    LabelPile = new ArrayList<String>();
	            for(int x = 0; x < things.size(); x++) LabelPile.add(TuiTeX.emph(things.get(x).label));
	  	    TuiTeX.enumerate("Inventory", LabelPile);	
                    break;
	       
                case 3:
                    LabelPile = new ArrayList<String>();
	            for(int x = 0; x < things.size(); x++) LabelPile.add(TuiTeX.emph(things.get(x).label));
                    NumMenu DiscardMenu = new NumMenu(LabelPile, "Choose a [Thing] to Discard: ");
                    things.remove(DiscardMenu.getSelection(true) - 1);
		    break;

                case 4:
	 	    System.exit(1);
                                 
	        default:
	   }
	   TuiTeX.vspace();
        }
    }
}

File: ThingTest/ThingTest.pl
#!/usr/bin/perl

#import java.util.*;
#
#public class ThingTest
#{
#    public static void main(String[] args)
#    {
#        System.out.print(TuiTeX.box("Things"));
system(qq(java mybanner "Things"));
#
#	ArrayList<Thing> things = new ArrayList<Thing>();
#
#	ArrayList<String> options = new ArrayList<String>();
$options[0] = "Add a [Thing] to your pile.";
$options[1] = "View your pile.";
$options[2] = "Exit";

#	options.add("Add a [Thing] to your pile.");
#	options.add("View your pile.");
#	options.add("Exit");
#
#        NumMenu menu = new NumMenu(options);
#
#        while(true){			
#	    switch(menu.getSelection()){
#
#            case 1:
#	       things.add(new Thing());
#               break;		    		
#
#	    case 2:
#	       ArrayList<String> LabelPile = new ArrayList<String>();
#	       for(int x = 0; x < things.size(); x++){
#	           LabelPile.add(TuiTeX.emph(things.get(x).getLabel()));
#
#	  	} 
#	       TuiTeX.enumerate("Inventory", LabelPile);	
#   break;
#	       
#
#            case 3:
#		System.exit(1);
#                break;
#             
#
#	      
#	       default:
#	    }
#	    TuiTeX.vspace();
#        }
#    }
#}
